pkg/tool/helm.go

```diff
@@ -2,82 +2,167 @@ package tool
 
 import (
 	"fmt"
+	"os"
+	"time"
 
-	"github.com/helm/chart-testing/v3/pkg/tool"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	"k8s.io/helm/pkg/strvals"
 )
 
-// Helm is an interface to the helm binary; it is a thin layer on top of the Helm abstraction offered by chart-testing
-// to silence output being streamed to Stdout.
 type Helm struct {
-	tool.Helm
-	ProcessExecutor
-	extraArgs []string
+	config   *action.Configuration
+	settings *cli.EnvSettings
+	args     map[string]interface{}
 }
 
-func NewHelm(exec ProcessExecutor, extraArgs []string) Helm {
-	return Helm{
-		tool.NewHelm(exec.ProcessExecutor, extraArgs),
-		exec,
-		extraArgs,
+func NewHelm(settings *cli.EnvSettings, args map[string]interface{}) (*Helm, error) {
+	helm := new(Helm)
+	helm.settings = settings
+	helm.args = args
+	config := new(action.Configuration)
+	if err := config.Init(settings.RESTClientGetter(), settings.Namespace(), os.Getenv("HELM_DRIVER"), func(format string, v ...interface{}) {
+		LogInfo(fmt.Sprintf(format, v))
```

Comment:
I believe that errors should not be logged in constructors, so let's just return the error here and deal with logging in the caller.

Reply:
In general yes I do agree, however in this case it's part of the Helm design decision to add a `DebugLog` function as one of the parameters. So I would incline to remain this specific logging, please let me know if I missed anything. WDYT?

Reply:
Oh, I overlooked and understood `LogInfo` was the first branch of the `if` statement.